apps/src/blockly/themes/accessibleStylesGenerator.mjs

```diff
@@ -0,0 +1,152 @@
+    color01Primary: '#460B70',
+    color01Secondary: '#00306F',
+    color02Primary: '#6B069F',
+    color02Secondary: '#00489E',
+    color03Primary: '#8E06CD',
+    color03Secondary: '#005FCC',
+    color04Primary: '#B40AFC',
+    color04Secondary: '#0079FA',
+    color05Primary: '#ED0DFD',
+    color06Primary: '#5F0914',
+    color06Secondary: '#004002',
+    color07Primary: '#86081C',
+    color07Secondary: '#005A01',
+    color08Primary: '#B20725',
+    color08Secondary: '#007702',
+    color09Primary: '#DE0D2E',
+    color09Secondary: '#009503',
+    color10Primary: '#FF4235',
+    color11Primary: '#5A0A33',
+    color11Secondary: '#003D30',
+    color12Primary: '#810D49',
+    color12Secondary: '#005745',
+    color13Primary: '#AB0D61',
+    color13Secondary: '#00735C',
+    color14Primary: '#D80D7B',
+    color14Secondary: '#009175',
+    color15Primary: '#FF2E95'
+  },
+  [themes.TRITANOPIA]: {
+    color01Primary: '#004002',
+    color01Secondary: '#003D30',
+    color01Tertiary: '#00306F',
+    color02Primary: '#005A01',
+    color02Secondary: '#005745',
+    color02Tertiary: '#00489E',
+    color03Primary: '#007702',
+    color03Secondary: '#00735C',
+    color03Tertiary: '#005FCC',
+    color04Primary: '#009503',
+    color04Secondary: '#009175',
+    color05Primary: '#5F0914',
+    color05Secondary: '#5A0A33',
+    color06Primary: '#86081C',
+    color06Secondary: '#810D49',
+    color07Primary: '#B20725',
+    color07Secondary: '#AB0D61',
+    color08Primary: '#DE0D2E',
+    color08Secondary: '#D80D7B',
+    color09Primary: '#FF4235',
+    color09Secondary: '#FF2E95',
+    color10Primary: '#460B70',
+    color11Primary: '#6B069F',
+    color12Primary: '#8E06CD',
+    color13Primary: '#B40AFC',
+    color14Primary: '#ED0DFD',
+    color15Primary: '#0079FA'
+  }
+};
+
+Object.values(themes).forEach(theme => {
+  mapBlockStylesToPalette(
+    {...JSON.parse(JSON.stringify(cdoBlockStyles))},
+    theme
+  );
+});
+
+console.log(
+  '\x1b[33m%s\x1b[0m',
+  'Copy the following values into cdoAccessible.js:'
+);
+
+function mapBlockStylesToPalette(blockStyles, theme) {
+  const enoughColors =
+    Object.keys(accessiblePalettes[theme]).length >=
+    Object.keys(blockStyles).length;
+
+  if (enoughColors) {
+    // Each color in our standard palette is mapped to a new "nearest" color from the accesible palette.
+    for (const [, value] of Object.entries(blockStyles)) {
+      const nearestAvailableColor = nearestColor.from(
+        accessiblePalettes[theme]
+      );
+      const newColor = nearestAvailableColor(value.colourPrimary);
+      value.colourPrimary = newColor.value;
+      const colorKey = newColor.name.substring(0, 7);
+      // Remove the color (and any alternates) from the available palette so we don't assign it twice.
+      delete accessiblePalettes[theme][`${colorKey}Primary`];
+      delete accessiblePalettes[theme][`${colorKey}Secondary`];
+      delete accessiblePalettes[theme][`${colorKey}Tertiary`];
```

Comment:
nit: it can be valuable with this kind of algorithm to think if there could be a way to achieve it but by leaving the source data structures unmodified.

Reply:
I suppose we could also just iterate over a new color "candidate" to ensure that it hasn't already been claimed. In this case, it seems like that would slow things down (perhaps negligibly) and result in the same final re-mapping.